cmd/nitro/nitro.go

```
@@ -834,79 +812,22 @@ func ParseNode(ctx context.Context, args []string) (*NodeConfig, *genericconf.Wa
-		"l2.chain-id":                        421613,
-		"init.empty":                         true,
-	}, "."), nil)
-}
-
-func applyArbitrumRollupRinkebyTestnetParameters(k *koanf.Koanf) error {
-	return k.Load(confmap.Provider(map[string]interface{}{
-		"persistent.chain":                   "rinkeby-nitro",
-		"node.forwarding-target":             "https://rinkeby.arbitrum.io/rpc",
-		"node.feed.input.url":                "wss://rinkeby.arbitrum.io/feed",
-		"l1.rollup.bridge":                   "0x85c720444e436e1f9407e0c3895d3fe149f41168",
-		"l1.rollup.inbox":                    "0x578BAde599406A8fE3d24Fd7f7211c0911F5B29e",
-		"l1.rollup.rollup":                   "0x71c6093c564eddcfaf03481c3f59f88849f1e644",
-		"l1.rollup.sequencer-inbox":          "0x957c9c64f7c2ce091e56af3f33ab20259096355f",
-		"l1.rollup.validator-utils":          "0x0ea7372338a589e7f0b00e463a53aa464ef04e17",
-		"l1.rollup.validator-wallet-creator": "0x237b8965cebe27108bc1d6b71575c3b070050f7a",
-		"l1.rollup.deployed-at":              11088567,
-		"l2.chain-id":                        421611,
-	}, "."), nil)
-}
-
-func applyArbitrumAnytrustGoerliTestnetParameters(k *koanf.Koanf) error {
-	return k.Load(confmap.Provider(map[string]interface{}{
-		"persistent.chain": "goerli-anytrust",
-	}, "."), nil)
+func applyChainParameters(k *koanf.Koanf, chainId uint64, l1ChainId uint64, l2ChainInfoFiles []string) (bool, error) {
+	chainInfo, err := chaininfo.ProcessChainInfo(chainId, l2ChainInfoFiles)
+	if err != nil {
+		return false, err
+	}
+	if chainInfo != nil && chainInfo.ChainParameters != nil {
+		if chainInfo.ParentChainId != l1ChainId {
+			return false, fmt.Errorf("ParentId: %d provided in %s for chainId: %d is not equal to l1ChainId: %d provided in commandline", chainInfo.ParentChainId, l2ChainInfoFiles, chainId, l1ChainId)
+		}
+		err = k.Load(rawbytes.Provider(*chainInfo.ChainParameters), json.Parser())
+		if err != nil {
+			return false, err
+		}
+		return true, nil
+	}
+	return false, nil
```

I think we should just return an error here if we don't find a chain, now that you're unable to specify everything a chain needs via the config directly

---

done.